pkg/dronereceiver/factory.go

```diff
@@ -2,53 +2,68 @@ package dronereceiver
 
 import (
 	"context"
-	"time"
 
 	"github.com/grafana/grafana-ci-otel-collector/dronereceiver/internal/metadata"
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/receiver"
 	"go.opentelemetry.io/collector/receiver/scraperhelper"
 )
 
+const (
+	defaultBindEndpoint = "0.0.0.0:3333"
+	defaultPath         = "/drone/webhook"
+)
+
 func createDefaultConfig() component.Config {
 	cfg := scraperhelper.NewDefaultScraperControllerSettings(metadata.Type)
-	cfg.CollectionInterval = 10 * time.Second
 
 	return &Config{
 		ScraperControllerSettings: cfg,
 		MetricsBuilderConfig:      metadata.DefaultMetricsBuilderConfig(),
-		WebhookConfig: WebhookConfig{
-			Endpoint: "/drone/webhook",
-			Port:     3333,
+		ServerConfig: confighttp.ServerConfig{
+			Endpoint: defaultBindEndpoint,
 		},
+		Path:   defaultPath,
+		Secret: "",
 	}
 }
 
 func NewFactory() receiver.Factory {
 	return receiver.NewFactory(
 		metadata.Type,
 		createDefaultConfig,
-		receiver.WithTraces(createTraceReceiver, metadata.TracesStability),
-		receiver.WithMetrics(createMetricsReceiver, metadata.MetricsStability),
-		receiver.WithLogs(createLogsReceiver, metadata.LogsStability))
+		receiver.WithTraces(newTracesReceiver, metadata.TracesStability),
+		receiver.WithMetrics(newMetricsReceiver, metadata.MetricsStability),
+		receiver.WithLogs(newLogsReceiver, metadata.LogsStability))
 }
 
-func createTraceReceiver(_ context.Context, set receiver.CreateSettings, cfg component.Config, consumer consumer.Traces) (receiver.Traces, error) {
-	r, err := getOrAddReceiver(set, cfg)
+func newTracesReceiver(_ context.Context, set receiver.CreateSettings, cfg component.Config, consumer consumer.Traces) (receiver.Traces, error) {
+	rCfg := cfg.(*Config)
+	var err error
+
+	if consumer == nil {
+		return nil, component.ErrNilNextConsumer
+	}
+
+	r := receivers.GetOrAdd(cfg, func() component.Component {
+		var rcv component.Component
+		rcv, err = newReceiver(set, rCfg)
+		return rcv
+	})
 	if err != nil {
 		return nil, err
 	}
 
-	r.Unwrap().enableTraces(consumer)
+	r.Unwrap().(*droneReceiver).tracesConsumer = consumer
 	return r, nil
-
 }
 
-func createMetricsReceiver(_ context.Context, set receiver.CreateSettings, rConf component.Config, consumer consumer.Metrics) (receiver.Metrics, error) {
+func newMetricsReceiver(_ context.Context, set receiver.CreateSettings, rConf component.Config, consumer consumer.Metrics) (receiver.Metrics, error) {
 	cfg := rConf.(*Config)
 	ns := newDroneScraper(set, cfg)
-	scraper, err := scraperhelper.NewScraper(metadata.Type, ns.scrape, scraperhelper.WithStart(ns.start))
+	scraper, err := scraperhelper.NewScraper(string(metadata.Type), ns.scrape, scraperhelper.WithStart(ns.start))
 
 	if err != nil {
```

Comment:
should we check if the consumer is empty here as well?

Reply:
the plan was to remove metrics from the receiver anyway in a follow up PR (to be more in line with the direction other CI/CD related receivers are going) so i didn't really touch anything around metrics, but yep, makes sense. will change 👍

Reply:
done in b2bb8d6